test/com/google/enterprise/adaptor/database/ResponseGeneratorTest.java

```diff
@@ -409,6 +599,36 @@ public void testBlobColumnModeContentTypeCol() throws Exception {
   }
 
   @Test
+  public void testBlobColumnModeContentTypeCol_nullValue() throws Exception {
+    String content = "hello world";
+    executeUpdate(
+        "create table data(id int, content blob, contentType varchar)");
+    String sql =
+        "insert into data(id, content) values (1, ?)";
+    try (PreparedStatement ps = getConnection().prepareStatement(sql)) {
+      ps.setBytes(1, content.getBytes(UTF_8));
+      assertEquals(1, ps.executeUpdate());
+    }
+
+    MockResponse bar = new MockResponse();
+    Response response = newProxyInstance(Response.class, bar);
+    Map<String, String> cfg = new TreeMap<String, String>();
+    cfg.put("columnName", "content");
+    cfg.put("contentTypeOverride", ""); // Empty should be ignored.
+    cfg.put("contentTypeCol", "contentType");
+    ResponseGenerator resgen = ResponseGenerator.contentColumn(cfg);
+
+    ResultSet rs = executeQueryAndNext("select * from data");
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(ResponseGenerator.class,
+        "content type at col {0} is null", messages);
+    resgen.generateResponse(rs, response);
+    assertEquals(content, bar.baos.toString(UTF_8.name()));
+    assertEquals(null, bar.contentType);
```

Comment:
s/assertEquals(null, /assertNull(/g? (The better error message here was fixed by JUnit 4.11.)